jlibra-core/src/main/java/dev/jlibra/admissioncontrol/transaction/Transaction.java

```diff
@@ -33,4 +45,60 @@ default ByteSequence serialize() {
                 .toByteSequence();
     }
 
+    public static Transaction fromGrpcObject(types.TransactionOuterClass.Transaction grpcTransaction) {
```

Comment:
I don't know yet how I feel about having so much code in `abstract class`/`interface`. What do you think about creating separate class `TransactionFactory`? Name is irrelevant but with creation responsibility.

Reply:
Yes, it might make sense to get the conversion / serialization out of the value objects. I will try something and see if how it starts to look like.

Reply:
The factory pattern could be a good solution for this. They neither abstract nor interface would implement logic which might actually change in implementation classes. But this might also just "move" the problem to the factory classes.